sdk/cosmos/azure-cosmos/tests/test_session_token_unit.py

```diff
@@ -86,6 +105,41 @@ def test_validate_session_token_comparison(self):
                              "Status code: 500\nCompared session tokens '1#101#1=20#2=5#3=30' "
                              "and '1#100#1=20#2=5#3=30#4=40' have unexpected regions.")
 
+        # same version with different region progress should throw error
+        session_token1 = VectorSessionToken.create("1#101#1=20#2=5#3=30")
+        session_token2 = VectorSessionToken.create("1#100#4=20#2=5#3=30")
+        self.assertIsNotNone(session_token1)
+        self.assertIsNotNone(session_token2)
+
+        try:
+            session_token1.merge(session_token2)
+            self.fail("Region progress can not be different when version is same")
+        except CosmosHttpResponseError as e:
+            self.assertEqual(str(e),
+                             "Status code: 500\nCompared session tokens '1#101#1=20#2=5#3=30' "
+                             "and '1#100#4=20#2=5#3=30' have unexpected regions.")
+
+    def test_validate_session_token_comparison(self):
+        self._different_merge_scenarios()
+        os.environ["AZURE_COSMOS_SESSION_TOKEN_FALSE_PROGRESS_MERGE"] = "false"
+        self._different_merge_scenarios()
+        del os.environ["AZURE_COSMOS_SESSION_TOKEN_FALSE_PROGRESS_MERGE"]
+
+    def test_session_token_false_progress_merge(self):
```

Comment:
Could you add a success scenario wherein a new region is added, or an existing region is removed (assume vector clock number incremented too)?